applications/ShallowWaterApplication/custom_elements/conserved_var_element.hpp

```diff
@@ -23,84 +23,266 @@
+
+    typedef typename BaseType::IndexType                              IndexType;
+
+    typedef typename BaseType::GeometryType                        GeometryType;
+
+    typedef typename BaseType::GeometryType::Pointer        GeometryPointerType;
+
+    typedef typename BaseType::PropertiesType                    PropertiesType;
+
+    typedef typename BaseType::PropertiesType::Pointer    PropertiesPointerType;
+
+    typedef typename BaseType::NodesArrayType                    NodesArrayType;
+
+    typedef typename BaseType::ElementVariables                ElementVariables;
+
+    typedef typename BaseType::EquationIdVectorType        EquationIdVectorType;
+
+    typedef typename BaseType::DofsVectorType                    DofsVectorType;
+
+    typedef typename BaseType::VectorType                            VectorType;
+
+    typedef typename BaseType::MatrixType                            MatrixType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
 
     /// Default constructor.
     ConservedVarElement() :
-        Element()
+        BaseType()
     {}
-    
-    ConservedVarElement(IndexType NewId, GeometryType::Pointer pGeometry) :
-        Element(NewId, pGeometry)
+
+    /// Constructor using a Geometry instance
+    ConservedVarElement(IndexType NewId, GeometryPointerType pGeometry) :
+        BaseType(NewId, pGeometry)
     {}
-    
-    ConservedVarElement(IndexType NewId, GeometryType::Pointer pGeometry, PropertiesType::Pointer pProperties) :
-        Element(NewId, pGeometry, pProperties)
+
+    /// Constructor using geometry and properties
+    ConservedVarElement(IndexType NewId, GeometryPointerType pGeometry, PropertiesPointerType pProperties) :
+        BaseType(NewId, pGeometry, pProperties)
     {}
 
     /// Destructor.
-    virtual ~ ConservedVarElement() {};
+    virtual ~ ConservedVarElement() override {};
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
 
-//----------------------------------------------------------------------
+    /// Create a new Primitive variables element and return a pointer to it
+    Element::Pointer Create(IndexType NewId, NodesArrayType const& rThisNodes, PropertiesPointerType pProperties) const override
+    {
+        KRATOS_TRY
+        return boost::make_shared< ConservedVarElement < TNumNodes > >(NewId, this->GetGeometry().Create(rThisNodes), pProperties);
+        KRATOS_CATCH("")
+    }
 
-    Element::Pointer Create(IndexType NewId, NodesArrayType const& ThisNodes,  PropertiesType::Pointer pProperties) const
+    Element::Pointer Create(IndexType NewId, GeometryPointerType pGeom, PropertiesPointerType pProperties) const override
     {
         KRATOS_TRY
-        return Element::Pointer(new ConservedVarElement(NewId, GetGeometry().Create(ThisNodes), pProperties));
+        return boost::make_shared< ConservedVarElement < TNumNodes > >(NewId, pGeom, pProperties);
```

Comment:
Change boost:: for Kratos::

Reply:
I am getting an error with Kratos::make_shared

Reply:
changed